File: epi_judge_python/h_index.py
from typing import List

from test_framework import generic_test


def h_index(citations: List[int]) -> int:
    citations.sort(reverse=True)
    h_index = 0
    for c in citations:
        if h_index + 1 > c:
            break
        h_index += 1
    return h_index


if __name__ == '__main__':
    exit(generic_test.generic_test_main('h_index.py', 'h_index.tsv', h_index))

File: epi_judge_cpp/h_index.cc
#include <algorithm> 
#include <vector>

#include "test_framework/generic_test.h"

using std::vector;


int HIndex(vector<int> citations) {
  int hIndex {0};
  std::sort(citations.begin(), citations.end(), std::greater<>());
  for (auto const & c : citations) {
    if (hIndex + 1 > c) {
      break;
    }
    ++hIndex;
  }
  return hIndex;
}


int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"citations"};
  return GenericTestMain(args, "h_index.cc", "h_index.tsv", &HIndex,
                         DefaultComparator{}, param_names);
}
